### ISSUE TITLE
Ensure compatability with hummingbird

### ISSUE THREAD

**Issue by adam2392** [neurodata/treeple#316]
Can we have an example of converting decision tree models to hummingbird for faster inference?

https://github.com/microsoft/hummingbird

In addition, an integration test would be great. 
1. Add hummingbird as a test dependency
2. Fit every forest we have. 
3. Convert to hummingbird
4. Verify inference on test set is numerically the same. 

Perhaps Ryan or Vlad can take this?

**Comment by ryanhausen** [comment]
@adam2392 

Vlad and I looked into this and have found a couple things:

1. Supporting our models would be somewhat involved. We need to do two things:
  -  rewrite our models so that can be converted by Hummingbird
  - add the converters to the hummingbird package by PR or maintain our own fork
  - More details here: https://github.com/microsoft/hummingbird/issues/707
2. That being said, humming bird does seem to offer some benefits the treeple package might be interested in. In particular, inference and on-disk storage.

I ran the following script to evaluate the inference speed and storage size on disk: [bench_hb.py.txt](https://github.com/user-attachments/files/17385329/bench_hb.py.txt)

Some of the combinations error out, but seems to be known issue: (https://github.com/microsoft/hummingbird/issues/666)

Below are some parallel coordinates plots that show performance as a function: # estimators, # features, # samples. The last axis is the thing being measured and is what is used for the colormap as well.  The metrics are the log10 of the ratio of sklearn to hummingbird. Blue is good for hummingbird, indicating that sklearn needs more time/storage. Red the opposite.

![cpu_time](https://github.com/user-attachments/assets/847b43a6-f8ee-4d52-a039-c8ba4c486b21)

![gpu_time](https://github.com/user-attachments/assets/3271590b-1c46-4b3c-ac09-2bc1dad89630)

![storage](https://github.com/user-attachments/assets/64f794ef-1c32-4e4e-b5b2-4c892448c51e)

In summary, it seems helpful if we want to store and reuse a model but might take a non-trivial amount of effort to implement and maintain.
